### DISCUSSION TITLE
Trouble running behind nginx reverse proxy (on different host)

### DISCUSSION THREAD

**Discussion by calharding** [searx/searx#2752] [Installation issue: Docker]
I am trying to run searx-docker on a host sitting behind a reverse proxy. running `start.sh` reports no problems or issues. On my reverse proxy, I have the following entry:

```
server {

        server_name searx.mydomain.tld;
        set $upstream 10.4.20.120;
        location / {
                proxy_pass http://$upstream;
        }
}
```

When I browse to `http://searx.mydomain.tld`, I can see that, somewhere, my browser is being redirected to `http://10.4.20.120` - which obviously times out after a while, not being publicly routable.

I strongly doubt that the redirection is being caused by my reverse proxy, because the entry is in the exact same format as a number of other services I have running on hosts behind it (same subnet), so that part is fine. The redirection must be happening somehow on the searx-docker host (Caddy maybe? I don't know Caddy well enough to fiddle with it, but I have not edited its configuration at all). I have changed the `SEARX_HOSTNAME` parameter in the .env file, but to no avail.

Any ideas?

**Comment by aidanharris** [comment]
This happens because your reverse proxy (Nginx) doesn't send the Host header correctly so searx uses `10.4.20.120` as the host name. A minimal reverse proxy config could look like this:

```nginx
server {

        server_name searx.mydomain.tld;
        set $upstream 10.4.20.120;
        location / {
                proxy_set_header Host $http_host;
                proxy_set_header X-Forwarded-Proto $scheme;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;

                proxy_pass http://$upstream;
        }
}
```
